src/document_basic.c

```diff
@@ -244,9 +245,7 @@ int Redis_SaveDocument(RedisSearchCtx *ctx, Document *doc, int options, QueryErr
 }
 
 int Document_ReplyFields(RedisModuleCtx *ctx, Document *doc) {
-  if (!doc) {
-    return REDISMODULE_ERR;
-  }
+  assert(doc);
```

Comment:
are we sure it will never happened?

Reply:
Yes.
It is called in two places and Document_LoadAllFields() is checked for errors before calling Document_ReplyFields()